examples/change_notifier_custom_state/lib/main.dart

```diff
@@ -145,21 +139,19 @@ class BooksLocation extends BeamLocation<BooksState> {
   BooksLocation(RouteInformation routeInformation) : super(routeInformation);
 
   @override
-  BooksState createState(RouteInformation routeInformation) =>
-      BooksState().fromRouteInformation(routeInformation);
+  BooksState createState(RouteInformation routeInformation) {
+    final state = BooksState().fromRouteInformation(routeInformation)
+                  ..addListener(notifyListeners);
+    return state;
+  }
+     
 
   @override
   void initState() {
     super.initState();
     state.addListener(notifyListeners);
   }
 
-  @override
-  void updateState(RouteInformation routeInformation) {
-    final booksState = BooksState().fromRouteInformation(routeInformation);
-    state.updateWith(booksState.isBooksListOn, booksState.selectedBookId);
-  }
-
```

Comment:
This is in fact important to update the state when navigation is done typing the URL directly into browser
For example
1. Type `/books/2` into browser URL, press enter
2. Pop books details
3. Click on a book from list
Nothing happens...